clientconn.go

```diff
@@ -719,6 +719,16 @@ func (cc *ClientConn) GetState() connectivity.State {
 	return cc.csMgr.getState()
 }
 
+// GetStatus returns the status of cc as an error produced by the status
+// package.  If the state is TransientFailure, the status error will be
+// non-nil.  Otherwise, it will always be nil.
+func (cc *ClientConn) GetStatus() error {
+	if cc.GetState() != connectivity.TransientFailure {
+		return nil
+	}
+	return status.Errorf(codes.Unavailable, "latest connection error: %v", cc.blockingpicker.connectionError())
```

Comment:
What's the advantage of using status error here?
What other error codes will we return?

Reply:
It mirrors the Java API.  That one returns a status directly, but I thought "error" would be more idiomatic for us.